### PR TITLE
SEARCH-627 & SEARCH-628 (Memory Indexing)

### PR DESCRIPTION
**Pull request by keerthi16**
## Description
Real-Time indexing (Memory)
[SEARCH-627](https://perzoinc.atlassian.net/browse/SEARCH-627)
Batch-Indexing (Memory)
[SEARCH-628](https://perzoinc.atlassian.net/browse/SEARCH-628)

## Approach
How does this change address the problem?
- Batch indexing to memory 
- Real-time index to memory.
- Search results from memory.

## Learning
N/A

#### Blog Posts
N/A

branch | PR
------ | ------
other_pr_rc | [link](https://github.com/SymphonyOSF/SFE-Client-App/pull/8744)
other_pr_dev | [link](https://github.com/SymphonyOSF/SFE-Client-App/pull/8743)

## Open Questions if any and Todos
- [x] Unit-Tests
screen shot 2018-03-06 at 12 35 04 pm
- [x] Documentation
- [ ] Automation-Tests
When solved, check the box and explain the answer.

@VikasShashidhar @serkanmulayim Please review. Thanks 😃

### CODE REVIEW

File: demo/search.html [hunk 1]
```diff
@@ -239,8 +240,9 @@ <h1>Symphony Electron Search API Demo</h1>
     });
 
     merge.addEventListener('click', function () {
-        search.mergeIndexBatches().then(function () {
+        search.ramIndexToFSIndex().then(function () {
```
Comment (VikasShashidhar) [inline demo/search.html]: Change this to `memoryIndexToFSIndex`

File: js/search/search.js [hunk 2]
```diff
@@ -65,16 +66,18 @@ class Search {
      * and creates a folder in the userData
      */
     init() {
+        libSymphonySearch.symSEDestroy();
         libSymphonySearch.symSEInit();
+        libSymphonySearch.symSEClearMainRAMIndex();
```
Comment (VikasShashidhar) [inline js/search/search.js]: Change all names containing `RAM` to `memory`

File: tests/Search.test.js [hunk 3]
```diff
@@ -78,12 +81,15 @@ describe('Tests for Search', function() {
         }, 3000);
     });
 
-    function deleteIndexFolders(location) {
+    function deleteIndexFolders(location, isEncryption) {
```
Comment (VikasShashidhar) [inline tests/Search.test.js]: Is the actual API inaccessible here? Why are we repeating it here?

**Comment by VikasShashidhar** [comment]
@keerthi16 There are too many changes in this PR. Can you break this down to different PRs?

**Comment by keerthi16** [comment]
@NitinSingh2020  conflicts resolved 😄

**Comment by keerthi16** [comment]
Please do not merge this PR. Thanks 😃

**Pull request merged**